### PR TITLE
Feature/capture read

### PR DESCRIPTION
**Pull request by AlonZivony**
## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [ ] Git log contains summary of the change.
- [ ] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

Capture file reads in addition to the file write capture.
Added capture filters for file captures to allow more specific capturing to maintain the great load of read operations.

Fixes: #2319

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [ ] I have made corresponding changes to the documentation.
- [ ] My code follows the style guidelines (C and Go) of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented all functions/methods created explaining what they do.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix, or feature, is effective.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [ ] Subject starts with "subsystem|file: description".
- [ ] Do not end the subject line with a period.
- [ ] Limit the subject line to 50 characters.
- [ ] Separate subject from body with a blank line.
- [ ] Use the imperative mood in the subject line.
- [ ] Wrap the body at 72 characters.
- [ ] Use the body to explain what and why instead of how.

### CODE REVIEW

File: pkg/ebpf/config.go
Comment (yanivagman) [inline pkg/ebpf/config.go:17]: Nice! We have issue #3031 to extract config package out of the ebpf package, and I think this takes us almost there

File: pkg/events/events.go
Comment (yanivagman) [inline pkg/events/events.go:5982]: These events overload the system and it might be that the user will choose to only capture, without having these events.
Although I know they are needed for the "written_files" to work - we need to reconsider adding them as a dependancy.

File: pkg/events/events.go
Comment (yanivagman) [inline pkg/events/events.go:6007]: Where do we use that?

Comment (AlonZivony) [reply pkg/events/events.go:6007]: We need it to know if a file is a pipe or not for the file type filter.
https://github.com/aquasecurity/tracee/pull/2356/files#diff-c046893996f2e2a6e44a8caac7a14abbf47b47486ee5ceaa9029c34fc5f94a99R393

File: pkg/ebpf/config.go
Comment (AlonZivony) [inline pkg/ebpf/config.go:70]: Unfinished comment

File: pkg/ebpf/c/tracee.bpf.c
Comment (AlonZivony) [reply pkg/ebpf/c/tracee.bpf.c:2692]: No, I think I missed this one

File: pkg/ebpf/c/common/filesystem.h
Comment (yanivagman) [inline pkg/ebpf/c/common/filesystem.h:353]: The READ_KERN macro was removed.
You should now use BPF_CORE_READ to do all this at once - like done here: https://github.com/aquasecurity/tracee/blob/main/pkg/ebpf/c/tracee.bpf.c#L4861

**Review by yanivagman — CHANGES_REQUESTED** [review]
Rebase is required.
Please note that READ_KERN macro was removed, see comment below.

**Review by yanivagman — APPROVED** [review]
LGTM
Will wait with the merge for @rafaeldtinoco to review

**Comment by rafaeldtinoco** [comment]
There are a few nits I've seen (wrong comments) but I have not had time to review it. Since we got a +1 from Yaniv, I'm merging this and will review after merged (mostly for self-knowledge about it and chores).

**Pull request merged**
